pymongo/collection.py

```diff
@@ -3080,8 +3082,7 @@ def find_one_and_update(self, filter, update,
             document matches the query. Defaults to ``False``.
           - `return_document`: If
             :attr:`ReturnDocument.BEFORE` (the default),
-            returns the original document before it was updated, or ``None``
-            if no document matches. If
```

Comment:
Can you keep this part? I think it's useful to keep the `None` return here too because this doc string is so large.

Reply:
I can, but then the description is asymmetric, describing the `None` return value if no document matches for `BEFORE` but not describing the return value for `AFTER`, which once again makes it unclear what the return value is for `AFTER`. By omitting this statement here, the user is directed to the statement above which applies to both `BEFORE` and `AFTER`.

In order to be consistent, one would also need to describe something similar for the `AFTER`, which makes the docstring even more verbose and redundant.

I'll follow your advice to illustrate what I mean.

Reply:
Thanks for pointing that out. I agree that we can revert your latest commit but I think it still needs one change to fix the asymmetry. The docs in the parent commit say:

> By default :meth:`find_one_and_update` returns the original version of
the document before the update was applied, or ``None`` if no document
matches the filter. To return the updated (or inserted in the case of
*upsert*), version of the document instead, use the *return_document*
option.

The asymmetry is still present in this paragraph: it describes `None` for `BEFORE` but not for `AFTER` just like before this change. Can you amend this section to mention when `None` is returned with `AFTER`?

Also can you remove the comma here: `*upsert*), version`?

Reply:
> Also can you remove the comma here?

Yes, easy.

> The asymmetry is still present in this paragraph.

Hmm. I had written it to mean "By default find_one_and_update returns None if no document matches the filter" (independent of "by default find_one_and_update returns the original version of the document before the update was applied"), even though they are in the same sentence. But I can see how your reading betrays my intention. I'll hit it again.